torch_geometric/nn/models/explainer.py

```diff
@@ -0,0 +1,113 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torch_geometric.nn import MessagePassing
+
+
+def set_masks(model: torch.nn.Module, mask: Tensor, edge_index: Tensor,
+              apply_sigmoid: bool = True):
+    """Apply mask to every graph layer in the model."""
+    loop_mask = edge_index[0] != edge_index[1]
+
+    # Loop over layers and set masks on MessagePassing layers
+    for module in model.modules():
+        if isinstance(module, MessagePassing):
+            module.__explain__ = True
+            module.__edge_mask__ = mask
+            module.__loop_mask__ = loop_mask
+            module.__apply_sigmoid__ = apply_sigmoid
+
+
+def clear_masks(model: torch.nn.Module):
+    """Clear all masks from the model."""
+    for module in model.modules():
+        if isinstance(module, MessagePassing):
+            module.__explain__ = False
+            module.__edge_mask__ = None
+            module.__loop_mask__ = None
+            module.__apply_sigmoid__ = True
+    return module
+
+
+class CaptumModel(torch.nn.Module):
+    r"""Model with forward function that can be easily used for
+    explainability with `Captum.ai <https://captum.ai/>`_.
+
+    Args:
+        model (torch.nn.Module): Model to be explained.
+        mask_type (str): Denotes the type of mask to be created with a Captum
+            explainer. Valid inputs are :obj:`'edge'`, :obj:`'node'`, and
+            :obj:`'node_and_edge'`. The input for the forward function with
+            mask_type :obj:`'edge'` should be an edge_mask tensor of shape
+            (1, num_edges), :obj:`x` and :obj:`edge_index`. The input for the
+            forward function with mask_type :obj:`'node'` should be a
+            node_input of shape (1, num_nodes, num_features) and
+            :obj:`edge_index`. The input for the forward function with
+            mask_type :obj:`'node_and_edge'` should be a node_input tensor of
+            shape (1, num_nodes, num_features), an edge_mask tensor of shape
+            (1, num_edges), and :obj:`edge_index`. (default: :obj:`'edge'`)
```

Comment:
Add a note that additional arguments are applicable as long as the order of the first arguments is guaranteed?